src/component/FeatureInfoGrid/FeatureInfoGrid.tsx

```diff
@@ -24,144 +26,98 @@ import _upperFirst from 'lodash/upperFirst';
-
-  downloadGridData: any; // Property to download grid data
-
+  hoverVectorLayer: OlLayerVector;
+  isTimeLayer?: boolean;
+  downloadGridData: boolean;
   onPaginationChange?: (idx: number) => void;
+  t: (arg: any) => string;
+}
 
-  /**
-   * Translate function
-   */
-  t: (arg: any) => void;
+interface ColumnDef {
+  headerName: string;
+  field: string;
+  minWidth: number;
+  cellRenderer?: (text: string) => string | React.ReactElement<any, string | React.JSXElementConstructor<any>>;
 }
 
-interface FeatureInfoGridState {
-  currentPage: number;
-  selectedFeat: OlFeature;
-  gridApi: any;
+interface RowData {
+  id: number | string;
+  attr: string;
+  val: string;
 }
 
+export type ComponentProps = DefaultFeatureInfoGridProps & FeatureInfoGridProps;
+
+let gridApi: GridApi;
+
 /**
  * Class representing a feature info grid.
  *
  * @class The FeatureInfoGrid.
  * @extends React.Component
  */
-export class FeatureInfoGrid extends React.Component<FeatureInfoGridProps, FeatureInfoGridState> {
-
-  /**
-   * The default properties.
-   */
-  public static defaultProps: DefaultFeatureInfoGridProps = {
-  };
-
-  /**
-   * Create a feature info grid component.
-   * @constructs FeatureInfoGrid
-   */
-  constructor(props: FeatureInfoGridProps) {
-    super(props);
-
-    props.features[0].set('selectedFeat', true);
-
-    this.state = {
-      selectedFeat: props.features[0],
-      currentPage: 1,
-      gridApi: {}
-    };
-
-    // binds
-    this.updateSize = this.updateSize.bind(this);
-    this.onPaginationChange = this.onPaginationChange.bind(this);
-    this.getTimeFeatureColumnDefs = this.getTimeFeatureColumnDefs.bind(this);
-    this.getTimeFeatureRowData = this.getTimeFeatureRowData.bind(this);
-    this.downloadData = this.downloadData.bind(this);
-    this.onGridIsReady = this.onGridIsReady.bind(this);
-  }
-
-  /**
-   * The componentDidUpdate function
-   *
-   * @param {FeatureInfoGridProps} prevProps Previous props.
-   * @param {FeatureInfoGridState} prevState Previous state.
-   */
-  componentDidUpdate(prevProps: FeatureInfoGridProps, prevState: FeatureInfoGridState) {
-    const {
-      features,
-      downloadGridData
-    } = this.props;
-
-    const {
-      selectedFeat
-    } = this.state;
-
-    if (!_isEqual(prevProps.features, features)) {
-      features[0].set('selectedFeat', true);
-      this.setState({
-        selectedFeat: features[0],
-        currentPage: 1
-      });
-    }
-    if (!_isEqual(prevState.selectedFeat, selectedFeat)) {
-      this.updateVectorLayer(selectedFeat);
+export const FeatureInfoGrid: React.FC<ComponentProps> = ({
+  downloadGridData,
+  features,
+  hoverVectorLayer,
+  isTimeLayer = false,
+  map,
+  onPaginationChange,
+  t
+}): JSX.Element => {
+
+  const [selectedFeat, setSelectedFeat] = useState<OlFeature>(features[0]);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  useEffect(() => {
+    features[0].set('selectedFeat', true);
```

Comment:
Is `features[0]` always available?

Reply:
I think we are safe here. Features are coming from the context menu already grouped by feature types, for which the GetFeatureInfo request has returned at least one result. Empty results won't be listed there at all.

But an additional check surely doesn't hurt :)

Reply:
Added in 34fb41f